From 5f63961294e22363ec4c0e31abbf737a35316baa Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Fri, 30 Oct 2020 18:35:57 +0800
Subject: [PATCH] ASoC: es8311: fix filling others bits

The width of ADC PGA gain is 4 and we need to mask them.

Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
Change-Id: I8b8f76a7feb4fa9c9e2066385f73035c1797730d
---
 sound/soc/codecs/es8311.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/es8311.c b/sound/soc/codecs/es8311.c
index dad0152e5033..e53854b2024e 100644
--- a/sound/soc/codecs/es8311.c
+++ b/sound/soc/codecs/es8311.c
@@ -573,8 +573,8 @@ static int es8311_probe(struct snd_soc_component *component)
 		snd_soc_component_update_bits(component, ES8311_GPIO_REG44,
 					      0x70, es8311->aec_mode << 4);
 	if (es8311->adc_pga_gain)
-		snd_soc_component_write(component, ES8311_SYSTEM_REG14,
-					es8311->adc_pga_gain);
+		snd_soc_component_update_bits(component, ES8311_SYSTEM_REG14,
+					      0x0f, es8311->adc_pga_gain);
 	if (es8311->adc_volume)
 		snd_soc_component_write(component, ES8311_ADC_REG17,
 					es8311->adc_volume);
-- 
2.35.3

